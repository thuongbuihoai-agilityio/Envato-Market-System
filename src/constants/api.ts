export const USERS_API = 'https://65660396eb8bb4b70ef2c750.mockapi.io';

// TODO: Will provide latter
export const BASE_API = '';
export const STATISTICAL_API = 'https://succinct-shrouded-acorn.glitch.me';

export const END_POINTS = {
  AUTHEN_USERS: '/authenUsers',
  USERS: '/users',
  TRANSACTIONS: '/transactions',
  STATISTICS: '/statistics',
};

export const SEARCH_PARAM = {
  EMAIL: 'email',
  PASSWORD: 'password',
};
