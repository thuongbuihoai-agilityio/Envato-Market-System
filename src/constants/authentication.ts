export const SUPER_ADMIN = 'Super Admin';

export const MEMBER = 'member';
