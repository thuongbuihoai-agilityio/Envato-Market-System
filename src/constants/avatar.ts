export enum AVATAR_POSITION {
  BEFORE = 'before',
  AFTER = 'after',
}
