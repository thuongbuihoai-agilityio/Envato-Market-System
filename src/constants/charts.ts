export const STROKE_COLORS = ['#22C55E', '#EAB308', '#EDF2F7', '#23262B'];
export const REVENUE_FLOW_COLORS = ['#EAB308', '#22C55E', '#FF784B', '#EDF2F7'];

export const MONTHS = [
  'Jan',
  'Feb',
  'Mar',
  'April',
  'May',
  'Jun',
  'July',
  'Aug',
  'Sep',
  'Oct',
  'Nov',
  'Dec',
];

export const REVENUE_FLOW_STATUS = ['Pending', 'Signed', 'Lost'];
