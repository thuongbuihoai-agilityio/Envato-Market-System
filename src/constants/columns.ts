// Types
import { TDataSource } from '@app/interfaces';

export const COLUMNS_DASHBOARD = (
  onRenderHead: (title: string, key: string) => void,
  onRenderBody: ({ id, image, name }: TDataSource) => void,
  onRenderActionIcon: (data: TDataSource) => void,
) => [
  {
    title: 'Customer name',
    key: 'name',
    renderHead: onRenderHead,
    renderBody: onRenderBody,
  },
  {
    title: 'Email',
    key: 'email',
    renderHead: onRenderHead,
  },
  {
    title: 'Location',
    key: 'location',
    renderHead: onRenderHead,
  },
  {
    title: 'Spent',
    key: 'spent',
    renderHead: onRenderHead,
  },
  {
    title: '',
    key: 'action',
    renderBody: onRenderActionIcon,
    renderHead: onRenderHead,
  },
];

export const COLUMNS_HISTORY = (
  renderHead: (title: string, key: string) => void,
  renderBody: ({ id, image, name }: TDataSource) => void,
  renderPaymentStatus: ({ paymentStatus }: TDataSource) => void,
  renderTransactionStatus: ({ paymentStatus }: TDataSource) => void,
  renderActionIcon: (data: TDataSource) => void,
) => [
  {
    title: 'Customer name',
    key: 'name',
    renderHead: renderHead,
    renderBody: renderBody,
  },
  {
    title: 'Date',
    key: 'date',
    renderHead: renderHead,
  },
  {
    title: 'Amount',
    key: 'spent',
    renderHead: renderHead,
  },
  {
    title: 'Payment',
    key: 'paymentStatus',
    renderHead: renderHead,
    renderBody: renderPaymentStatus,
  },
  {
    title: 'Status',
    key: 'transactionStatus',
    renderHead: renderHead,
    renderBody: renderTransactionStatus,
  },
  {
    title: '',
    key: 'action',
    renderBody: renderActionIcon,
  },
];
