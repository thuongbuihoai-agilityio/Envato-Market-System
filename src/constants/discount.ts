export const DEFAULT_DISCOUNT_PERCENTAGE = 60;
export const LIMIT_OF_BONUS = 5;
