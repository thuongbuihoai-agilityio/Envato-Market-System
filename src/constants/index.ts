export * from './routers';
export * from './time';
export * from './images';
export * from './form';
export * from './messages';
export * from './regex';
export * from './texts';
export * from './themes';
export * from './api';
export * from './images';
export * from './menu';
export * from './sidebar';
export * from './charts';
export * from './sidebar';
export * from './common';
export * from './options';
export * from './transaction';
export * from './store';
export * from './status';
export * from './pagination';
export * from './columns';
