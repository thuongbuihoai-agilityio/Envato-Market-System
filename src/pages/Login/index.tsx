import { FormEvent, memo, useMemo } from 'react';
import { ViewOffIcon, ViewIcon } from '@chakra-ui/icons';
import { Button, HStack, Text, VStack, useDisclosure } from '@chakra-ui/react';
import { Link } from 'react-router-dom';

// Constants
import { ROUTES } from '@constants/index';

// Layouts
import { AuthLayout } from '@layouts/index';

// Components
import { InputField, Checkbox } from '@components/index';

const LoginPage = (): JSX.Element => {
  const { isOpen: isShowPassword, onToggle: onToggleShowPassword } =
    useDisclosure();

  const renderPasswordIcon = useMemo((): JSX.Element => {
    const Icon = isShowPassword ? ViewIcon : ViewOffIcon;

    return (
      <Icon
        color="gray.400"
        w="25px"
        h="25px"
        cursor="pointer"
        onClick={onToggleShowPassword}
      />
    );
  }, [isShowPassword, onToggleShowPassword]);

  return (
    <AuthLayout>
      <VStack
        as="form"
        gap={4}
        mb={6}
        onSubmit={(e: FormEvent) => e.preventDefault()}
      >
        <InputField
          variant="authentication"
          placeholder="Username or email"
          onChange={() => {}} // TODO: Will update when API integrate
        />
        <InputField
          type={isShowPassword ? 'text' : 'password'}
          variant="authentication"
          placeholder="Password"
          rightIcon={renderPasswordIcon}
          onChange={() => {}} // TODO: Will update when API integrate
        />
      </VStack>

      {/* Helpers */}
      <HStack justifyContent="space-between" w="100%">
        <Checkbox
          variant="round"
          onChange={
            () => {} // TODO: Will update when API integrate
          }
        >
          <Text fontWeight="semibold">Remember me</Text>
        </Checkbox>
        <Text
          as={Link}
          to={ROUTES.FORGOT_PASSWORD}
          color="primary.500"
          fontWeight="semibold"
          textTransform="capitalize"
          textDecoration="underline"
        >
          forgot password?
        </Text>
      </HStack>

      <Button textTransform="capitalize" my={7}>
        Sign In
      </Button>
      <Text fontWeight="medium" textAlign="center">
        Don&apos;t have an account?{' '}
        <Text
          as={Link}
          to={ROUTES.REGISTER}
          fontWeight="semibold"
          textDecoration="underline"
        >
          Sign Up
        </Text>
      </Text>
    </AuthLayout>
  );
};

const Login = memo(LoginPage);
export default Login;
