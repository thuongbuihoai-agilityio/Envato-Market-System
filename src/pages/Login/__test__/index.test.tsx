import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { act, render, screen, waitFor } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

// Components
import Login from '@app/pages/Login';

// Constants
import { ERROR_MESSAGES, ROUTES } from '@app/constants';

const setup = () =>
  render(
    <QueryClientProvider client={new QueryClient()}>
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    </QueryClientProvider>,
  );

describe('Login page', () => {
  it('Match to snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it('should handle submit correctly', async () => {
    const { container } = setup();

    await act(async () => {
      const usernameInput = screen.getByRole<HTMLInputElement>('textbox');

      const passwordInput =
        screen.getByPlaceholderText<HTMLInputElement>(/password/i);

      const checkboxLabel = screen.getByText(/remember me/i);

      await userEvent.type(usernameInput, 'dieu.le@asnet.com.vn');

      await userEvent.type(passwordInput, 'Abcd@123');

      await userEvent.click(checkboxLabel);
    });

    const currentUrl = global.window.location.pathname;

    expect(currentUrl).toContain(ROUTES.LOGIN);

    const submitBtn = container.querySelector('button[type="submit"]');
    submitBtn && (await userEvent.click(submitBtn));

    expect(currentUrl).toContain(ROUTES.ROOT);
  });

  it('should handle submit failed', async () => {
    setup();

    const usernameInput = screen.getByRole<HTMLInputElement>('textbox');

    const passwordInput =
      screen.getByPlaceholderText<HTMLInputElement>(/password/i);

    const checkboxLabel = screen.getByText(/remember me/i);

    const submitBtn = screen.getByRole('button', {
      name: /sign in/i,
    });

    await userEvent.type(usernameInput, 'dieu.le@asnet.com.vn');

    await userEvent.type(passwordInput, 'Abcd@12356345');

    await userEvent.click(checkboxLabel);

    await userEvent.click(submitBtn);

    waitFor(async () => {
      const message = screen.findByText(
        new RegExp(ERROR_MESSAGES.AUTH_INCORRECT, 'i'),
      );
      expect(message).toEqual(ERROR_MESSAGES.AUTH_INCORRECT);
    });
  });
});
