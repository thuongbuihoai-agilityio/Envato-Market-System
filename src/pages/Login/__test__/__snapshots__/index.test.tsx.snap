// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page Match to snapshot 1`] = `
<div>
  <div
    class="css-hboir5"
  >
    <section
      class="css-1rv26q0"
    >
      <a
        class="chakra-text css-bljho8"
        href="/"
      >
        <img
          alt="This is logo"
          class="chakra-image css-0"
          src="/icons/logo.svg"
        />
      </a>
      <div
        class="css-qeb7js"
      >
        <header
          class="chakra-stack css-tl3ftk"
        >
          <h1
            class="chakra-heading css-owo776"
          >
            Sign in to Bankco.
          </h1>
          <p
            class="chakra-text css-1xbgeyr"
          >
            Send, spend and save smarter
          </p>
        </header>
        <div
          class="css-1wqp5dg"
        >
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-ri09he"
          />
          <div
            class="css-1u70y5c"
          >
            Or continue with
          </div>
        </div>
        <form
          class="chakra-stack css-ylms7g"
          id="login-form"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <div
              class="chakra-input__group css-169ufh8"
              data-group="true"
            >
              <input
                class="chakra-input css-0"
                id="field-:r1:"
                name="username"
                placeholder="Username or email"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <div
              class="chakra-input__group css-169ufh8"
              data-group="true"
            >
              <input
                class="chakra-input css-0"
                id="field-:r2:"
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
              <button
                class="chakra-input__right-element css-1uzzgp1"
              >
                <svg
                  class="chakra-icon css-1l546ul"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="currentColor"
                  >
                    <path
                      d="M23.2,10.549a20.954,20.954,0,0,0-4.3-3.6l4-3.995a1,1,0,1,0-1.414-1.414l-.018.018a.737.737,0,0,1-.173.291l-19.5,19.5c-.008.007-.018.009-.026.017a1,1,0,0,0,1.631,1.088l4.146-4.146a11.26,11.26,0,0,0,4.31.939h.3c4.256,0,8.489-2.984,11.051-5.8A2.171,2.171,0,0,0,23.2,10.549ZM16.313,13.27a4.581,4.581,0,0,1-3,3.028,4.3,4.3,0,0,1-3.1-.19.253.253,0,0,1-.068-.407l5.56-5.559a.252.252,0,0,1,.407.067A4.3,4.3,0,0,1,16.313,13.27Z"
                    />
                    <path
                      d="M7.615,13.4a.244.244,0,0,0,.061-.24A4.315,4.315,0,0,1,7.5,12,4.5,4.5,0,0,1,12,7.5a4.276,4.276,0,0,1,1.16.173.244.244,0,0,0,.24-.062l1.941-1.942a.254.254,0,0,0-.1-.421A10.413,10.413,0,0,0,12,4.75C7.7,4.692,3.4,7.7.813,10.549a2.15,2.15,0,0,0-.007,2.9,21.209,21.209,0,0,0,3.438,3.03.256.256,0,0,0,.326-.029Z"
                    />
                  </g>
                </svg>
              </button>
            </div>
          </div>
          <div
            class="chakra-stack css-qo2j2d"
          >
            <label
              class="chakra-checkbox css-1elxtiq"
            >
              <input
                class="chakra-checkbox__input"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value=""
              />
              <span
                aria-hidden="true"
                class="chakra-checkbox__control css-dnty2r"
              />
              <span
                class="chakra-checkbox__label css-1oeb2oe"
              >
                <p
                  class="chakra-text css-10ircq3"
                >
                  Remember me
                </p>
              </span>
            </label>
            <a
              class="chakra-text css-csnbyf"
              href="/forgot-password"
            >
              forgot password?
            </a>
          </div>
        </form>
        <button
          class="chakra-button css-8o2rc7"
          form="login-form"
          type="submit"
        >
          Sign In
        </button>
        <p
          class="chakra-text css-25kmip"
        >
          Don't have an account?
           
          <a
            class="chakra-text css-13g7tb6"
            href="/register"
          >
            Sign Up
          </a>
        </p>
        <div
          class="css-178yklu"
        >
          <ul
            class="css-1s6o0ok"
            role="list"
          >
            <li
              class="css-z9iakq"
            >
              <a
                class="chakra-link css-1bl9y41"
                href="#"
              >
                Terms & Conditions
              </a>
            </li>
            <li
              class="css-z9iakq"
            >
              <a
                class="chakra-link css-1bl9y41"
                href="#"
              >
                Privacy Policy
              </a>
            </li>
            <li
              class="css-z9iakq"
            >
              <a
                class="chakra-link css-1bl9y41"
                href="#"
              >
                Help
              </a>
            </li>
            <li
              class="css-z9iakq"
            >
              <a
                class="chakra-link css-1bl9y41"
                href="#"
              >
                English
              </a>
            </li>
          </ul>
          <p
            class="chakra-text css-e9tuiy"
          >
            @ 
            2023
             Bankco. All Rights Reserved.
          </p>
        </div>
      </div>
    </section>
    <div
      class="css-1nyw6b"
    >
      <img
        alt="This is sign in image"
        class="chakra-image css-0"
        src="/icons/signin.svg"
      />
      <img
        alt="This is square image"
        class="chakra-image css-nuaex1"
        src="/icons/square.svg"
      />
      <img
        alt="This is vline image"
        class="chakra-image css-bvz8js"
        src="/icons/vline.svg"
      />
      <img
        alt="This is dotted image"
        class="chakra-image css-12wsc5z"
        src="/icons/dotted.svg"
      />
      <div
        class="css-1wwt0r"
      >
        <h3
          class="chakra-heading css-tmruzz"
        >
          Speady, Easy and Fast
        </h3>
        <p
          class="chakra-text css-1rphqkz"
        >
          BankCo. help you set saving goals, earn cash back offers, Go to
          disclaimer for more details and get paychecks up to two days early. Get
          a
            
          <span
            class="text-highlight"
          >
            
              $20
            
          </span>
          
          bonus when you receive qualifying direct deposits
        </p>
      </div>
    </div>
  </div>
</div>
`;
